### DISCUSSION TITLE
Why the need for the codewars testing framework?

### DISCUSSION THREAD

**Discussion by FranzMartyn** [codewars/codewars.com#3268] [Q&A]
Codewars offers 2 testing frameworks for Python:

- `unittest`
- `codewars_test`

What's the need for creating a seperate testing framework? Does `unittest` not offer some features that would be a lot more helpful for creating and translating katas? Do kata creators and translators prefer using a JUnit-like testing framework? Or is there something else?

**Comment by hobovsky** [comment]
Slightly related: https://github.com/codewars/runner/issues/50. Note that `unittest` is available in kumite, but not in kata.

**Reply by FranzMartyn** [reply]
@kazk points out [here](https://github.com/codewars/runner/issues/50#issuecomment-421167544) that there can be only one testing framework per language due to the database. Is the issue still there?

**Reply by hobovsky** [reply]
AFAIK yes.
